skbio/stats/composition.py

```diff
@@ -608,10 +611,298 @@ def centralize(mat):
+    of zero cannot be computed.  While this is an unsolved problem, many
+    studies have shown promising results by replacing the zeros with pseudo
+    counts.
+
+    References
+    ----------
+    .. [1] Mandal et al. "Analysis of composition of microbiomes: a novel
+       method for studying microbial composition", Microbial Ecology in Health
+       & Disease, (2015), 26
+
+    Examples
+    --------
+    First import all of the necessary modules:
+
+    >>> from skbio.stats.composition import ancom
+    >>> import pandas as pd
+
+    Now let's load in a pd.DataFrame with sample and feature ids
+    for our data matrix:
+
+    >>> table = pd.DataFrame([[10., 11., 10., 10., 10., 10., 10.],
+    ...                       [10.5, 11.5, 10.5, 10.5, 10.5, 10.5, 10.5],
+    ...                       [10., 11., 10., 10., 10., 10., 10.],
+    ...                       [20., 21., 10., 10., 10., 10., 10.],
+    ...                       [20.5, 21.5, 10.5, 10.5, 10.5, 10.5, 10.5],
+    ...                       [20.3, 21.3, 10.2, 10.3, 10.1, 10.6, 10.4]],
+    ...                      index=['s1','s2','s3','s4','s5','s6'],
+    ...                      columns=['b1','b2','b3','b4','b5','b6','b7'])
+
+    Then create a grouping vector.  In this scenario, there
+    are only two classes, so the first three samples fall under the first
+    class while the last three samples fall under the second class:
+
+    >>> grouping = pd.Series([0, 0, 0, 1, 1, 1],
+    ...                      index=['s1','s2','s3','s4','s5','s6'])
+
+    Now run `ancom` and see if there are any features that have any
+    significant differences:
+
+    >>> results = ancom(table, grouping)
+    >>> results['W']
+    b1    6
+    b2    6
+    b3    2
+    b4    2
+    b5    2
+    b6    2
+    b7    2
+    Name: W, dtype: int64
+
+    The W-statistic is the number of features that a single feature is
+    tested to be significantly different against.  In this scenario
+    there are b1 was significantly different from all of the other features:
+
+    >>> results['reject']
+    b1     True
+    b2     True
+    b3    False
+    b4    False
+    b5    False
+    b6    False
+    b7    False
+    Name: reject, dtype: bool
+
+    Here, there were only two features that estimated to
+    significantly change between the two groups.
+
+    """
+    if len(table) != len(grouping):
+        raise ValueError('The number of samples in table must be equal '
+                         '(%d samples) to the number of samples in '
+                         'grouping (%d samples).' % (len(table),
+                                                     len(grouping)))
+
+    if multiple_comparisons_correction is not None:
+        if multiple_comparisons_correction != 'holm-bonferroni':
+            raise ValueError('%s is not an available option'
+                             % multiple_comparisons_correction)
+
+    if significance_test is None:
+        significance_test = scipy.stats.ttest_ind
```

Comment:
Change the default `scipy.stats.f_oneway`

Now, we'll need to be careful about comparing `ttest_ind` to `f_oneway`, because there are actually multiple types of t-tests encompassed in `ttest_ind`.  Anova assumes equal variance, whereas the Welch t-test doesn't.  So there are still scenarios when there are only 2 groups and it is preferable to use the Welch t-test.